main/src/ca/uwaterloo/flix/api/lsp/CodeActionContext.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 Holger Dal Mogensen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp
+
+import ca.uwaterloo.flix.util.Result
+import ca.uwaterloo.flix.util.Result.{Err, Ok}
+import org.json4s.JsonAST._
+
+object CodeActionContext {
+  def parse(json: JValue): Result[CodeActionContext, String] = {
+    val diagnosticsResult = json \ "diagnostics" match {
+      case JArray(l) => Result.traverse(l)(Diagnostic.parse)
+      case v => Err(s"Unexpected non-array diagnostics: '$v'.")
+    }
+
+    val onlyResult: Result[List[CodeActionKind], String] = json \ "only" match {
```

Comment:
1. I'm surprised we have to roll our own parser here. Does `json4s` not provide conversion-to-case-class facilities?
2. If not, I think we can abstract away some of the common functionality here, like `getOptionalString` and `getOptionalList` seem to be generally useful.

Reply:
We usually do roll our own parsers. My surprise is that w eneed to parse diagnostics.

But I think @sockmaster27 is just faithfully implementing the API.

But I suspect we can get away with something simpler.

Q: CodeActionContext is something sent _to us_ right?

Reply:
Yes

Reply:
If codeactioncontext is not sent to us, why do we need to parse it?

Reply:
It _is_ sent to us

Reply:
Removing parsing of `Diagnostic` pretty much eliminates this, so unless there are any objections I'll mark this as resolved

Reply:
Sounds good 👍